### ISSUE TITLE
Extending generic class results in stack overflow

### ISSUE THREAD

**Issue by armandino** [naver/fixture-monkey#546]
## Describe the bug
I've been playing around with some of the examples and ran into an issue extending a generic class.
Creating an instance of the subclass results in a stack overflow error.
Maybe I am missing some setup for this to work. If so, please let me know.

## Your environment
* version of Fixture Monkey: 0.4.11
* version of Java: 17

## Steps to reproduce
``` java
class Foo<T> {
    T foo;
}

class Bar extends Foo<String> {
    Double bar;
}

@Test
void example() {
    FixtureMonkey sut = FixtureMonkey.builder()
            .defaultGenerator(BeanArbitraryGenerator.INSTANCE)
            .build();

    Bar actual = sut.giveMeOne(Bar.class);

    assertThat(actual).isNotNull();
}
```

## Expected behaviour
Return an instance of the class

## Actual behaviour
```
java.lang.StackOverflowError
	at java.base/java.util.ArrayList.grow(ArrayList.java:244)
	at java.base/java.util.ArrayList.add(ArrayList.java:454)
	at java.base/java.util.ArrayList.add(ArrayList.java:467)
	at java.base/java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
	at org.junit.platform.commons.util.ReflectionUtils.findAllFieldsInHierarchy(ReflectionUtils.java:1204)
	at org.junit.platform.commons.util.ReflectionUtils.findFields(ReflectionUtils.java:1190)
	at com.navercorp.fixturemonkey.TypeSupports.extractFields(TypeSupports.java:92)
	at com.navercorp.fixturemonkey.arbitrary.ArbitraryTraverser.doTraverse(ArbitraryTraverser.java:111)
	at com.navercorp.fixturemonkey.arbitrary.ArbitraryTraverser.doTraverse(ArbitraryTraverser.java:140)
	at com.navercorp.fixturemonkey.arbitrary.ArbitraryTraverser.doTraverse(ArbitraryTraverser.java:140)
```

**Comment by seongahjo** [comment]
@armandino 
The example above is using Fixture Monkey 0.3
Fixture Monkey 0.3 is deprecated since it was not suitable to add complicated features.

You should use `FixtureMonkey.labMonkeyBuilder()` using 0.4
But there is a same issue in 0.4
I'll notice you when it is fixed

**Comment by armandino** [comment]
@seongahjo - thanks for letting me know that it's deprecated. I saw the example in `com.navercorp.fixturemonkey.test.ArbitraryGeneratorTest` in `0.4.11-SNAPSHOT`. Should it be marked as `@Deprecated` in the source?

**Comment by seongahjo** [comment]
@armandino 
Yes, it should be marked as deprecated!
Thank you for finding a mistake 👍

**Comment by seongahjo** [comment]
@armandino 
Thank you for making an issue.
It is fixed since 0.4.12.

**Comment by armandino** [comment]
@seongahjo - awesome! Thank you :)
